buildSrc/src/test/java/org/elasticsearch/gradle/plugin/PluginBuildPluginTests.java

```diff
@@ -0,0 +1,56 @@
+package org.elasticsearch.gradle.plugin;
+
+import org.elasticsearch.gradle.test.GradleUnitTestCase;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.testfixtures.ProjectBuilder;
+import org.junit.Before;
+
+import java.util.stream.Collectors;
+
+public class PluginBuildPluginTests extends GradleUnitTestCase {
+
+    private Project project;
+
+    @Before
+    public void setUp() throws Exception {
+        project = ProjectBuilder.builder()
+            .withName(getClass().getName())
+            .build();
+    }
+
+    public void testApply() {
+        project.getPlugins().apply(PluginBuildPlugin.class);
+
+        assertNotNull(
+            "plugin extension created with the right name",
+            project.getExtensions().findByName(PluginBuildPlugin.PLUGIN_EXTENSION_NAME)
+        );
+        assertNotNull(
+            "plugin extensions has the right type",
+            project.getExtensions().findByType(PluginPropertiesExtension.class)
+        );
+
+        assertNotNull(
+            "plugin created an integTest class",
+            project.getTasks().findByName("integTest")
+        );
+    }
+
+    public void testApplyWithAfterEvaluate() {
+        project.getPlugins().apply(PluginBuildPlugin.class);
+        PluginPropertiesExtension extension = project.getExtensions().getByType(PluginPropertiesExtension.class);
+        extension.setNoticeFile(project.file("test.notice"));
+        extension.setDescription("just a test");
+        extension.setClassname(getClass().getName());
+
+        PluginBuildPlugin.configureAfterEvaluate(project);
```

Comment:
I think the better thing to do here is simply call `Project.evaluate()`. It more accurately represents what will happen and will also trigger other `afterEvaluate` code that we haven't explicitly refactored in this way.

Reply:
There is no (longer) such method. I tried calling it with reflection too.
```
Method evaluate = project.getClass().getDeclaredMethod("evaluate");
evaluate.invoke(project);
```

Reply:
You'll have to cast it to `ProjectInternal`.

https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/java/org/gradle/api/internal/project/ProjectInternal.java#L65

Reply:
The cast works but then leads to a very strange cast exception down the road: https://gradle-enterprise.elastic.co/s/jjbp3xhyh3pvc/tests/td2ppmui4dyvs-7dxeqxtgf7z3g?openStackTraces=WzJd

Reply:
I suspect what is happening here is that actually evaluating the project is exposing some weird behavior. Probably because we have some dependencies between plugins that aren't being well defined. Sounds like the nebula `InfoJavaPlugin` is doing something weird.